.github/workflows/test-coverage.yml

```diff
@@ -0,0 +1,52 @@
+name: Nightly Test Coverage
```

Comment:
Why not just include this in the built and test workflow? Not sure why this needs to be a nightly job

Reply:
You are probably right, I was initially concerned about integration and unit tests running in different contexts, and then having to combine multiple test outputs into a final report... even with this though, different kernels could take different paths... we should do this in our existing infra.